internal/execute/table/stream.go

```diff
@@ -183,7 +183,7 @@ func (s *streamTable) Do(f func(flux.ColReader) error) error {
 	if err := f(s.first); err != nil {
 		s.first.Release()
 		s.first = nil
-		return nil
+		return err
```

Comment:
This seems to be a big omission. If the first buffer reports an error, stream seems to just discard it.